apps/agent-engine/cmd/agent-engine/serve.go

```
@@ -0,0 +1,303 @@
+	})
+	mux.HandleFunc("GET /health", func(w http.ResponseWriter, _ *http.Request) {
+		writeJSON(w, http.StatusOK, map[string]string{"status": "ok"})
+	})
+
+	// A leftover socket from a killed process would make Listen fail with
+	// EADDRINUSE even though nothing is serving it. The deploy restarts this
+	// daemon on every deploy, so that is the normal path, not the rare one.
+	if err := os.Remove(socketPath); err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("agent-engine: remove stale socket %s: %w", socketPath, err)
+	}
+	ln, err := net.Listen("unix", socketPath)
```

golangci-lint (2.12.2)

[error] net.Listen must not be called. use (*net.ListenConfig).Listen

(noctx)